tests/test_decision_maker/test_base.py

```diff
@@ -452,8 +539,91 @@ def test__is_utility_enhancing(self):
         self.decision_maker.ownership_state._quantities_by_good_id = None
         assert self.decision_maker._is_utility_enhancing(tx_message)
 
+    def test_sign_tx(self):
+        """Test the private function sign_tx of the decision maker."""
+        tx_hash = Web3.keccak(text="some_bytes")
+
+        tx_message = TransactionMessage(performative=TransactionMessage.Performative.PROPOSE_FOR_SIGNING,
+                                        skill_callback_ids=["default"],
+                                        tx_id="transaction0",
+                                        tx_sender_addr="agent_1",
+                                        tx_counterparty_addr="pk",
+                                        tx_amount_by_currency_id={"FET": -20},
+                                        tx_sender_fee=0,
+                                        tx_counterparty_fee=0,
+                                        tx_quantities_by_good_id={"good_id": 0},
+                                        ledger_id="fetchai",
+                                        info={'dialogue_label': "some_value"},
+                                        signing_payload={"tx_hash": tx_hash})
+
+        tx_signature = self.decision_maker._sign_tx(tx_message)
+        assert tx_signature is not None
+
+        tx_message = TransactionMessage(performative=TransactionMessage.Performative.PROPOSE_FOR_SIGNING,
+                                        skill_callback_ids=["default"],
+                                        tx_id="transaction0",
+                                        tx_sender_addr="agent_1",
+                                        tx_counterparty_addr="pk",
+                                        tx_amount_by_currency_id={"FET": -20},
+                                        tx_sender_fee=0,
+                                        tx_counterparty_fee=0,
+                                        tx_quantities_by_good_id={"good_id": 0},
+                                        ledger_id="off_chain",
+                                        info={'dialogue_label': "some_value"},
+                                        signing_payload={"tx_hash": tx_hash})
+
+        tx_signature = self.decision_maker._sign_tx(tx_message)
+        assert tx_signature is not None
+
     @classmethod
     def teardown_class(cls):
         """Tear the tests down."""
         cls._unpatch_logger()
         cls.multiplexer.disconnect()
+
+
+class Test_LedgerStateProxy:
+    """Test the Ledger State Proxy."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the test."""
+        cls.ledger_apis = LedgerApis({FETCHAI: DEFAULT_FETCHAI_CONFIG}, FETCHAI)
+        cls.ledger_state_proxy = LedgerStateProxy(ledger_apis=cls.ledger_apis)
+
+    def test_ledger_apis(self):
+        """Test the returned ledger_apis."""
+        assert self.ledger_state_proxy.ledger_apis == self.ledger_apis, "Must be equal."
+
+    def test_transaction_is_affordable(self):
```

Comment:
one method per test please and clearly describe in the docstring what exactly it is you are testing